storage/tokudb/ha_tokudb.cc

```diff
@@ -360,20 +360,40 @@ void TOKUDB_SHARE::set_cardinality_counts_in_table(TABLE* table) {
                 (i == table->s->primary_key) ||
                 (table->key_info[i].flags & HA_NOSAME);
 
-            uint32_t num_key_parts = get_key_parts(&table->key_info[i]);
-            for (uint32_t j = 0; j < num_key_parts; j++) {
+            KEY* key = &table->key_info[i];
+
+            /* Check if this index supports index statistics. */
+            if (!key->supports_records_per_key()) {
+                continue;
+            }
+
+            for (uint32_t j = 0; j < key->actual_key_parts; j++) {
+                if (j >= key->user_defined_key_parts) {
+                    key->set_records_per_key(j, 1.0);
+                    key->rec_per_key[j] = 1;
+                    continue;
+                }
+
                 assert_always(next_key_part < _rec_per_keys);
+                if ((key->flags & HA_FULLTEXT) || (key->flags & HA_SPATIAL)) {
+                    /* The record per key does not apply to
+                    FTS or Spatial indexes. */
```

Comment:
Good point, these are unnecessary in here until such point that we do support FULLTEXT and/or SPATIAL, if we ever do.